lib/rqrcode_core/qrcode/qr_util.rb

```diff
@@ -74,8 +74,8 @@ def self.get_bch_format_info(data)
     end
 
     def self.rszf(num, count)
-      # zero fill right shift
-      (num >> count) & ((2**((num.size * 8) - count)) - 1)
+      # right shift zero fill
+      (num >> count) & ((1 << (32 - count)) - 1)
```

Comment:
Moving my comments from the old PR thread to here. I *think* that all the numbers should be the same size, so you could add a constant to `QRUtils` like `ARCH_BITS = 1.size * 8` and just use that in the calculation to ensure you're always using the correct 32/64 bit value.

Reply:
I did actually try the constant approach. I've listed all the results below. I do wonder if the results as skewed due to me being on a `64` bit machine. If I hard code everything to 32 you can see that's when the improvement happens.

```
irb(main):002:0> 1.size * 8
=> 64

(num >> count) & ((1 << (32 - count)) - 1)
=> {:data=>"/workspaces/rqrcode_core/lib/rqrcode_core/qrcode/qr_util.rb", :count=>112240}

(num >> count) & ((1 << (64 - count)) - 1)
=> {:data=>"/workspaces/rqrcode_core/lib/rqrcode_core/qrcode/qr_util.rb", :count=>2882600}

ARCH_BITS = 1.size * 8
(num >> count) & ((1 << (ARCH_BITS - count)) - 1)
=> {:data=>"/workspaces/rqrcode_core/lib/rqrcode_core/qrcode/qr_util.rb", :count=>2882600}

(num >> count) & ((2**(32 - count)) - 1)
=> {:data=>"/workspaces/rqrcode_core/lib/rqrcode_core/qrcode/qr_util.rb", :count=>112240}

(num >> count) & ((2**((num.size * 8) - count)) - 1)
=> {:data=>"/workspaces/rqrcode_core/lib/rqrcode_core/qrcode/qr_util.rb", :count=>3699640}

(num >> count) & ((2**(ARCH_BITS - count)) - 1)
=> {:data=>"/workspaces/rqrcode_core/lib/rqrcode_core/qrcode/qr_util.rb", :count=>3699640}
```

Reply:
What are the implications to using `32` always? Performance?

Reply:
OK, from the comments I thought the issue was that running `num.size` over and over was causing the issue. This is just a wild guess, but I would assume the extra RAM comes from representing a number using 64 bits instead of 32? If so, I would think it shouldn't be an issue as long as we expect all numbers to fit within a 32 bit integer.